### ISSUE TITLE
When using scenarios, don't throw until test is finished.

### ISSUE THREAD

**Issue by vectro** [origin-energy/java-snapshot-testing#175]
If a test has multiple scenarios, we currently throw an exception the first time we call `toMatchSnapshot()` with different data. But then we miss the opportunity to update the rest of the scenarios in the .snap.debug file.

It would be better to let the test finish, and then raise if necessary with all differences together.

**Comment by jackmatt2** [comment]
Yes - agree that this is the preferred behaviour.  Would accept a PR implementing this.
